content/docs/tasks/traffic-management/secure-ingress-sds/index.md

```diff
@@ -0,0 +1,488 @@
+---
+title: Securing Gateways with SDS
+description: Describes how to configure Istio to expose a service outside of the service mesh, over TLS or Mutual TLS, using SDS.
+weight: 31
+keywords: [traffic-management,ingress,sds]
+---
+
+The [Securing Gateways with HTTPS](/docs/tasks/traffic-management/secure-ingress) task describes how to configure an ingress
+gateway to expose an HTTP endpoint of a service to external traffic with either simple or mutual TLS.
+This task shows you how to do the same, but using an ingress gateway agent to provision key/cert and root cert to ingress gateway dynamically.
+
+## Before you begin
+
+1.  Perform the steps in the [Before you begin](/docs/tasks/traffic-management/ingress#before-you-begin) and [Determining the ingress IP and ports](/docs/tasks/traffic-management/ingress#determining-the-ingress-ip-and-ports) sections of the
+[Control Ingress Traffic](/docs/tasks/traffic-management/ingress) task. After performing those steps you should have Istio and the [httpbin]({{< github_tree >}}/samples/httpbin) service deployed, and the environment variables `INGRESS_HOST` and `SECURE_INGRESS_PORT` set.
+
+1.  For macOS users, verify that you use _curl_ compiled with the [LibreSSL](http://www.libressl.org) library:
+
+    {{< text bash >}}
+    $ curl --version | grep LibreSSL
+    curl 7.54.0 (x86_64-apple-darwin17.0) libcurl/7.54.0 LibreSSL/2.0.20 zlib/1.2.11 nghttp2/1.24.0
+    {{< /text >}}
+
+    If a version of _LibreSSL_ is printed as in the output above, your _curl_ should work correctly with the
+    instructions in this task. Otherwise, try another installation of _curl_, for example on a Linux machine.
+
+## Configure a TLS ingress gateway with secret discovery service enabled
+
+In this section you will configure a TLS ingress gateway that fetches credentials from ingress gateway agent via secret discovery service (SDS).
+The ingress gateway agent is running in the same pod as the ingress gateway, and watches credentials created in the same namespace as the ingress gateway.
+Enabling SDS at ingress gateway brings the following benefits.
+
+* The ingress gateway is able to dynamically add/delete/update ingress gateway key/cert and root cert. You do not have to restart ingress gateway.
+
+* The secret volume mount is no longer needed. Once you create a `kubernetes` secret, that secret is captured by gateway agent and sent to ingress gateway as key/cert or root cert.
+
+* Gateway agent is able to watch multiple key/certs. You only need to create secrets for multiple hosts and update gateway definition.
+
+1.  Enable SDS at ingress gateway and deploy ingress gateway agent.
+    This feature is disabled by default, you need to enable the [feature flag]({{<github_blob>}}/install/kubernetes/helm/subcharts/gateways/values.yaml#L17) in helm,
```

Comment:
Please don't point users to specific lines in files, this will get out of date almost immediately.

Reply:
Done.